### DISCUSSION TITLE
can you have multiple users

### DISCUSSION THREAD

**Discussion by Nunya-9018** [coder/code-server#3420] [Q&A] [answered]
is there a way to have multiple users

**Comment by jsjoeio** [comment, marked as answer]
Could you elaborate on what you mean by "multiple users"? Do you mean multiple code-server instances running, the same instance but with multiple users connected, something else?

These posts may be related:
- https://github.com/cdr/code-server/discussions/3293
- https://github.com/cdr/code-server/discussions/2634
- https://github.com/cdr/code-server/discussions/2280

**Reply by Nunya-9018** [reply]
I mean is there a way to have multiple user profiles with there own user names and passwords and default settings

**Reply by jsjoeio** [reply]
No, there is nothing specifically built into code-server for that. There are some ideas [here](https://github.com/cdr/code-server/discussions/2280) though. Would love to hear more about your use case though and then we may be able to provide suggestions.

cc @bpmct

**Reply by Nunya-9018** [reply]
https://reposhub.com/javascript/editors/coderpair-vscode-live.html i found this but cant get it to work 
i am using pterodactyl panel

**Reply by jsjoeio** [reply]
Oh this is https://github.com/coderpair/vscode-live. I have not used it before so I don't think I can help there. Maybe someone else can though? cc @oxy @code-asher

**Reply by Nunya-9018** [reply]
yeah

**Reply by Nunya-9018** [reply]
> 
> 
> Oh this is https://github.com/coderpair/vscode-live. I have not used it before so I don't think I can help there. Maybe someone else can though? cc @oxy @code-asher

does it work with pterodactyl panel

**Reply by Nunya-9018** [reply]
does it eyes is not a answer

**Reply by jsjoeio** [reply]
I haven't used pterodactyl panel so the eyes meant I was checking it out. I am not sure if it works with it, sorry.

**Reply by Nunya-9018** [reply]
Okay

**Reply by bpmct** [reply]
coderpair is a fork of code-server and requires some additional configuration such as a firebase rtdb and configuration on a custom endpoint.

Because of this, it wouldn't work on a platform like pterodactyl out of the box, but could probably be automated with a significant amount of effort with the firebase API and modifying the config file for code-server for each instance of code-server created. It also hasn't been updated to the latest version, but the update process is documented.

@rdbeach may have more info on how this could work. I tried coderpair when it first came out, but it looks like it hasn't been updated to the latest version.

Are you still using it, and are there things we can do to make it easier to keep up to sync with code-server? I know we have a concept of "plugins" in code-server, but I believe it's not fully supported/documented yet.

**Comment by pys078** [comment]
you can use different ports to do so
